src/models/participant_model.py

```diff
@@ -62,14 +65,34 @@ def row(self):
     def removeChild(self, row):
         del self.childItems[row]
 
+    def removeChildByChild(self, child):
+        if child in self.childItems:
+            index = self.childItems.index(child)  # Get the index of the child
+            del self.childItems[index]  # Remove the child at the index
+
     def isDomain(self):
         return self.layer == DisplayLayerEnum.DOMAIN
 
+    def isParticipant(self):
+        return self.layer == DisplayLayerEnum.PARTICIPANT
+
+    def isTopic(self):
+        return self.layer == DisplayLayerEnum.TOPIC
+
+    def isReader(self):
+        return self.layer == DisplayLayerEnum.READER
+
+    def isWriter(self):
+        return self.layer == DisplayLayerEnum.WRITER
 
 class ParticipantTreeModel(QAbstractItemModel):
 
-    IsDomainRole = Qt.UserRole + 1
-    DisplayRole = Qt.UserRole + 2
+    DisplayRole = Qt.UserRole + 1
+    DomainRole = Qt.UserRole + 2
+    ParticipantRole = Qt.UserRole + 3
+    TopicRole = Qt.UserRole + 4
+    ReaderRole = Qt.UserRole + 5
+    WriterRole = Qt.UserRole + 6
```

Comment:
Hi @eboasson the two enums are there for different functionality, the `DisplayLayerEnum` defines what kind the nodes is. And the `Qt.UserRole` define which "roles" or variables of the model are gonna be available in the frontend (QML).
So they just sound kind of similar because in the frontend i want to check what kind of node i have to display a different text.
So all good.